pkg/packet/bgp/bgp.go

```diff
@@ -11623,6 +11624,98 @@ func NewTunnelEncapSubTLVColor(color uint32) *TunnelEncapSubTLVColor {
 	}
 }
 
+type TunnelEncapSubTLVEgressEndpoint struct {
+	TunnelEncapSubTLV
+	Address net.IP
+}
+
+func (t *TunnelEncapSubTLVEgressEndpoint) DecodeFromBytes(data []byte) error {
+	value, err := t.TunnelEncapSubTLV.DecodeFromBytes(data)
+	if err != nil {
+		return err
+	}
+	if t.Length != 6 && t.Length != 10 && t.Length != 22 {
+		return NewMessageError(BGP_ERROR_UPDATE_MESSAGE_ERROR, BGP_ERROR_SUB_MALFORMED_ATTRIBUTE_LIST, nil, "Not all TunnelEncapSubTLVEgressEndpoint bytes available")
+	}
+	address_family := binary.BigEndian.Uint16(value[4:6])
+	if address_family == 0 && t.Length != 6 || address_family == AFI_IP && t.Length != 10 || address_family == AFI_IP6 && t.Length != 22 {
+		return NewMessageError(BGP_ERROR_UPDATE_MESSAGE_ERROR, BGP_ERROR_SUB_MALFORMED_ATTRIBUTE_LIST, nil, "Not all TunnelEncapSubTLVEgressEndpoint bytes available")
+	}
+	if address_family == 0 {
```

Comment:
what about:
```
t.Address = nil
if address_family != 0 {
t.Address = blah
}
```
In this case you do not need if/else and it will be easier to read.

Reply:
I removed the else branch from this if statement and from two other similar if statements. I also was able to remove a switch statement with duplicated code.